### ISSUE TITLE
Operation request: Run a saved recipe

### ISSUE THREAD

**Issue by n1474335** [gchq/CyberChef#280]
### Summary

A new Flow Control operation that runs a saved recipe. The operation would have an argument that is a drop-down list of all currently saved recipes. When executed, it would run the saved recipe in place. This would effectively enable the use of functions.

The operation could be called "Run macro", "Run saved recipe", "Run function" or something similar.

**Comment by PenguinGeorge** [comment]
I'm currently taking a look at this.

Currently I've got it so that you have to paste the content of a recipe in to the input of the operation. Next I'll look in to the drop down of recipes as originally suggested.

I seem to be having issues with that though as I can't seem to access `localStorage`. Any ideas on this?

**Comment by n1474335** [comment]
Yeah, this one might be a little tricky. You can't access localStorage from a Web Worker (which all the operations run in). Perhaps we could create a new argument type which gets populated by the app itself? The worker doesn't need to know the names of the stored recipes in advance, it just needs to know the value of the recipe which is chosen when it gets sent through.

This could get a bit messy but I'd be interested to see a solution.

**Comment by PenguinGeorge** [comment]
I'll investigate further, as I think there is a good solution to this somewhere - just have to find it I guess!

**Comment by n1474335** [comment]
Another option is to pass the whole of localStorage through into the worker when it gets registered. That seems like a code smell though, and you'd have to keep passing it through again whenever the user makes any changes to anything in localStorage.

**Comment by PenguinGeorge** [comment]
I think your idea for a new argument type is better. I’ll look in to that next, then see about passing localStorage in to the worker if that fails. If there’s an elegant way to do that I’ll put it in to a PR for all to review.
